Creación del formulario de creación/edición de testimonios

gabrielpindo opened the pull request:
![image](https://user-images.githubusercontent.com/69809704/199739679-d3e2d1c9-30a7-40c8-bf4d-cdc9d02c9372.png)
![image](https://user-images.githubusercontent.com/69809704/199739765-be9efa29-1ed0-4a46-a33e-b0bf5886a0f2.png)

gabrielpindo committed:
Creación del Formulario Edicion y creacion de testimonios

gabrielpindo committed:
Creacion del formulario de edicion/creacion de testimonios

gabrielpindo committed:
Update TestimonialsForm.css

SoraBeat commented:
Falta borrar este comentario xd
![image](https://user-images.githubusercontent.com/30334118/199801044-cd840806-52fb-490f-a520-a7b829b849b9.png)
Falta validacion de imagen al ingresar un formato que no es png/jpg (me deja seleccionar un video y no salta error al submitear)
![image](https://user-images.githubusercontent.com/30334118/199800927-c9a833c7-da36-492d-8828-82bafecd2e03.png)
![image](https://user-images.githubusercontent.com/30334118/199801165-1a5b1f85-a839-4ed8-9333-b439ea080e24.png)

gabrielpindo committed:
Correcciones hechas (validación de imagen con yup y borrar comentario de ruta en App)

milandre commented on public/style.css:
```diff
@@ -6,6 +6,8 @@ Assigned to:    Andrew Noah
 Primary use:    ThemeDunk 
 -------------------------------------------------------------------*/
 
+
```
Eliminar espacios extra

milandre replied:
@gabrielpindo falta modificar aca

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,167 @@
+			axios
+				.get(`https://ongapi.alkemy.org/api/testimonials/${id}`)
+				.then(res => setDataTestimonials(res.data.data))
+				.catch(error => console.log(error));
```
Añadir sweetalert2 aca para los mensajes de error que sean consistentes con los otros formularios de la parte del backoffice

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,167 @@
+						if (!id) {
+							axios
+								.post(`https://ongapi.alkemy.org/api/testimonials`, values)
+								.then(res => console.log(res))
```
Añadir sweetalert2 aca para los mensajes de error que sean consistentes con los otros formularios de la parte del backoffice

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,167 @@
+							axios
+								.post(`https://ongapi.alkemy.org/api/testimonials`, values)
+								.then(res => console.log(res))
+								.catch(err => console.log(err));
```
Añadir sweetalert2 aca para los mensajes de error que sean consistentes con los otros formularios de la parte del backoffice

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,167 @@
+						} else {
+							axios
+								.put(`https://ongapi.alkemy.org/api/testimonials/${id}`, values)
+								.then(res => console.log(res))
```
Añadir sweetalert2 aca para los mensajes de error que sean consistentes con los otros formularios de la parte del backoffice

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,167 @@
+							axios
+								.put(`https://ongapi.alkemy.org/api/testimonials/${id}`, values)
+								.then(res => console.log(res))
+								.catch(err => console.log(err));
```
Añadir sweetalert2 aca para los mensajes de error que sean consistentes con los otros formularios de la parte del backoffice

milandre commented on src/App.js:
```diff
@@ -12,16 +13,17 @@ import MembersForm from './Components/Members/MembersForm'
           <Route path="/create-news" element={<NewsForm/>} />
           <Route path="/backoffice/create-slide" element={<SlidesForm/>} />
           <Route path="/create-testimonials" element={<TestimonialForm/>} />
+          <Route path="/create-testimonials/:id" element={<TestimonialForm/>} />
```
Como esto pertenece al backoffice, deberia tener una ruta con /backoffice/. Cambiar a `/backoffice/testimonials/:id
`

milandre replied:
@gabrielpindo falta hacer esto

milandre commented on src/App.js:
```diff
@@ -12,16 +13,17 @@ import MembersForm from './Components/Members/MembersForm'
           <Route path="/create-category" element={<CategoriesForm/>} />
           <Route path="/create-news" element={<NewsForm/>} />
           <Route path="/backoffice/create-slide" element={<SlidesForm/>} />
           <Route path="/create-testimonials" element={<TestimonialForm/>} />
```
Como esto pertenece al backoffice, deberia tener una ruta con /backoffice/. Cambiar a `/backoffice/testimonials`

milandre replied:
@gabrielpindo falta hacer esto

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,167 @@
+	};
+	return (
+		<div className="h-screen">
+			<div className="w-full sm:w-full sm:mx-auto md:w-1/2 lg:w-4/5 md:mx-auto flex flex-col justify-center items-center">
```
Los estilos y estructura de este formulario deben ser consistentes con los otros formularios que han sido creados para la parte del administrador. Sugiero reunirse con @SoraBeat, @YonaBordon, @DiegoSantelices10 para ponerse de acuerdo con esto.

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,167 @@
+									name="name"
+									onChange={handleChange}
+									defaultValue={dataTestimonials.name || ""}
+									placeholder="Ingresa el nombre de la categoria"
```
Deberia decir "Ingresa el nombre del testimonio"

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,167 @@
+									rounded-lg  hover:bg-sky-500 hover:-translate-y-1 
+									transition-all duration-500 text-white text-lg font-medium"
+									>
+										{!id ? "Subir testimonio" : "Actualizar testimonio"}
```
Cambiar "Subir testimonio" por "Crear testimonio"

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,167 @@
+						yup.object().shape({
+							name: yup.string().required(message),
+							description: yup.string().required(message),
+							image: yup.string().required().oneOf(["image/png", "image/jpg"],messageImg)
```
Revisar validaciones de imagen. Me sale esto y no me deja crear:
![image](https://user-images.githubusercontent.com/5572252/200669221-1db16d78-e69b-4231-abc6-3a8ca8d6a6c4.png)

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,167 @@
+			<div className="w-full sm:w-full sm:mx-auto md:w-1/2 lg:w-4/5 md:mx-auto flex flex-col justify-center items-center">
+				<Formik
+					initialValues={{
+						name: "",
```
¿Qué pasa si estoy en la vista de edición con los initialValues? Revisar esta parte. Trato de modificar solo la imagen para probar la actualización y me aparecen estos errores que no deberían estar
![image](https://user-images.githubusercontent.com/5572252/200669916-cf29c0ff-42f6-40ab-8789-a62a57203485.png)

milandre reviewed (changes requested):
@gabrielpindo TestimonialsForm.css es un archivo vacio que no deberia estar. Eliminarlo o en su defecto, colocar los estilos relacionados en ese archivo

gabrielpindo committed:
Update de <TestimonialsForm /> con estilos unificados

gabrielpindo committed:
Update

gabrielpindo committed:
Update con correcciones de estilo del componente para crear y editar Testimonios

milandre commented on src/Services/Testimonials/testimonials.js:
```diff
@@ -0,0 +1,31 @@
+	axios
+		.post(`https://ongapi.alkemy.org/api/testimonials`, values)
+		.then(res => {
+			console.log(res);
```
Borrar console.log

milandre commented on src/Services/Testimonials/testimonials.js:
```diff
@@ -0,0 +1,31 @@
+	axios
+		.put(`https://ongapi.alkemy.org/api/testimonials/${id}`, values)
+		.then(res => {
+			console.log(res);
```
Borrar console.log

milandre commented on src/Components/common/Form/Form.jsx:
```diff
@@ -1,14 +1,11 @@
-			className="  flex flex-col justify-center align-middle w-11/12 md:w-3/4 lg:w-2/4 h-screen gap-3 mx-auto"
-			onSubmit={handleSubmit}
-		>
+		<Form className=" lg:shadow rounded-md  w-full sm:w-4/5 md:w-11/12 lg:w-4/5  bg-white py-7">
```
Revisar el estilo en el PR #12 para que quede el formulario centrado

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,167 +1,127 @@
-										src={
-											values.image === "" ? dataTestimonials.image : values.image
-										}
+								<FormError />
```
Acá falta las props correspondientes a la imagen (error y touched)

gabrielpindo committed:
Correcciones hechas sobre el Componente de Edicion/creacion de testimonios

-Se borraron console.log()
-Se centro el formulario
-Se añadieron las props de error de "Image"

gabrielpindo commented:
Se realizaron los cambios necesarios.

milandre commented on src/Services/privateApiService.js:
```diff
@@ -2,7 +2,7 @@ import axios from 'axios';
 const config = {
     headers: {
-        Group: 01                //Aqui va el ID del equipo!!
+        Group: 1                //Aqui va el ID del equipo!!
```
Aca deberia ser grupo 4

milandre commented on src/Services/publicApiService.js:
```diff
@@ -2,7 +2,7 @@ import axios from 'axios';
 const config = {
     headers: {
-        Group: 01                //Aqui va el ID del equipo!!
+        Group: 1                //Aqui va el ID del equipo!!
```
Aca deberia ser grupo 4

milandre commented on src/Components/Testimonials/TestimonialsForm.js:
```diff
@@ -1,32 +1,131 @@
+	};
+
+	return (
+		<div className="w-full flex justify-center">
```
Falta centrarlo a lo vertical

gabrielpindo committed:
Correcciones realizadas

gabrielpindo committed:
Update App.js

gabrielpindo committed:
Merge branch 'dev' of https://github.com/alkemyTech/React-SXXI-T4 into features/ONG4-31/FormularioEdicionCreacionTestimonios

gabrielpindo committed:
Agregar rutas

gabrielpindo committed:
Merge branch 'dev' of https://github.com/alkemyTech/React-SXXI-T4 into features/ONG4-31/FormularioEdicionCreacionTestimonios

gabrielpindo committed:
Update ScreenData
